queries/roc/highlights.scm

```diff
@@ -0,0 +1,228 @@
+;----comments----
+[
+  (line_comment)
+  (doc_comment)
+] @comment
+
+;-----Punctuation----
+[
+  "?"
+  (arrow)
+  (back_arrow)
+  (backslash)
+] @punctuation.delimiter
+
+[
+  ","
+  ":"
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "{"
+  "}"
+  "["
+  "]"
+] @punctuation.bracket
+
+[
+  "|"
+  "&"
+  (operator)
+  (wildcard_pattern)
+] @operator
+
+[
+  "if"
+  "then"
+  "else"
+] @conditional
+
+[
+  (implements)
+  (when)
+  (is)
+  "as"
+  (to)
+] @keyword.control.roc
+
+;----headers-----
+(interface_header
+  (name) @type.interface)
+
+(imports
+  (imports_entry
+    (module) @namespace))
+
+(packages
+  (record_pattern
+    (record_field_pattern
+      (field_name) @namespace)))
+
+(app_name) @string
+
+(import_path) @string
+
+[
+  "app"
+  "packages"
+  "imports"
+  "provides"
+  "interface"
+  "exposes"
+  "expect"
+  (import_as)
+] @keyword.control
+
+;---annotations----
+(annotation_type_def
+  (annotation_pre_colon
+    (identifier) @type)
+  (#set! "priority" 99))
+
+(annotation_type_def
+  (annotation_pre_colon
+    (identifier) @function)
+  (function_type))
+
+;----decleration types----
+(value_declaration
+  (decl_left
+    (identifier_pattern
+      (identifier) @parameter.definition))
+  (#set! "priority" 99))
+
+(backpassing_expr
+  assignee: (identifier_pattern
+    (identifier) @parameter.definition))
+
+(value_declaration
+  (decl_left
+    (identifier_pattern
+      (identifier) @function))
+  (expr_body
+    (anon_fun_expr)))
+
+;----tags----
+(tags_type
+  (apply_type
+    (concrete_type) @constructor))
+
+(tag) @constructor
+
+(opaque_tag) @constructor
+
+;-----builtins----
+(variable_expr
+  (module) @module
+  (identifier) @constant.builtin.boolean
+  (#eq? @constant.builtin.boolean "true")
+  (#eq? @module "Bool"))
+
+(variable_expr
+  (module) @module
+  (identifier) @constant.builtin.boolean
+  (#eq? @constant.builtin.boolean "false")
+  (#eq? @module "Bool"))
+
+"dbg" @constant.builtin
+
+;----function invocations ----
+(function_call_expr
+  caller: (variable_expr
+    (identifier) @function.call))
+
+(function_call_expr
+  caller: (field_access_expr
+    (identifier) @function.call .))
+
+(bin_op_expr
+  (operator
+    "|>") @operator
+  (variable_expr
+    (identifier) @function))
+
+;----function arguments----
+(argument_patterns
+  (identifier_pattern
+    (identifier) @parameter))
+
+(argument_patterns
+  (_
+    (identifier_pattern
+      (identifier) @parameter)))
+
+(argument_patterns
+  (_
+    (_
+      (identifier_pattern
+        (identifier) @parameter))))
+
+(argument_patterns
+  (_
+    (_
+      (_
+        (identifier_pattern
+          (identifier) @parameter)))))
+
+(argument_patterns
+  (_
+    (_
+      (_
+        (_
+          (identifier_pattern
+            (identifier) @parameter))))))
+
+(argument_patterns
+  (_
+    (_
+      (_
+        (_
+          (_
+            (identifier_pattern
+              (identifier) @parameter)))))))
```

Comment:
Long story short: no
Why 5 depths, why not 6,7,8,9 depths is an example question. At worst please do queries like these that go as much as 2 depth, at best just the direct child

Reply:
Or use `#has-ancestor?`. (Expensive, but will be improved upstream, so it's better than copy-pasting variations on a theme.)